composer/trainer/trainer.py

```diff
@@ -415,25 +424,26 @@ def __init__(
         self,
         *,
         model: ComposerModel,
-        train_dataloader: Union[DataLoader, DataSpec],
-        max_duration: Union[int, str, Time],
-        eval_dataloader: Optional[Union[DataLoader, DataSpec, Evaluator, Sequence[Evaluator]]] = None,
-        algorithms: Optional[Union[Algorithm, Sequence[Algorithm]]] = None,
-        optimizers: Optional[torch.optim.Optimizer] = None,
-        schedulers: Optional[Union[ComposerScheduler, Sequence[ComposerScheduler]]] = None,
+        train_dataloader: Union[DataLoader, DataSpec],  # TODO move param to fit()
+        max_duration: Union[int, str, Time],  # TODO move param to fit()
+        eval_dataloader: Optional[Union[DataLoader, DataSpec, Evaluator,
+                                        Sequence[Evaluator]]] = None,  # TODO move param to fit()
+        algorithms: Optional[Union[Algorithm, Sequence[Algorithm]]] = None,  # TODO move param to fit()
+        optimizers: Optional[torch.optim.Optimizer] = None,  # TODO move param to fit()
+        schedulers: Optional[Union[ComposerScheduler, Sequence[ComposerScheduler]]] = None,  # TODO move param to fit()
```

Comment:
I am not in favor of moving all these parameters into `fit`, does not seem neccessary for the goal of enabling eval-only mode?

Reply:
Left arguments as-is for now; can revisit later.